Casks/n/nx-studio.rb

```diff
@@ -11,6 +11,23 @@
   livecheck do
     url "https://downloadcenter.nikonimglib.com/en/products/564/NX_Studio.html"
     regex(/Ver\.\s+(\d+(?:\.\d+)+)/i)
+    strategy :page_match do |page, regex|
+      version = page.scan(regex).filter_map { |match| match.first.tr(" ", "") }&.first
+
+      html_id = page.scan(%r{/download/sw/(\d+)\.html}i)&.first
+      next if html_id.blank?
+
+      version_page = Homebrew::Livecheck::Strategy.page_content("https://downloadcenter.nikonimglib.com/en/download/sw/#{html_id[0]}.html")
+      download_query = version_page[:content]&.match(/id="controllerMac".*redirect.do\?(.*?)[" <]/im)
+      next if download_query.blank?
+
+      redirect_url = "https://crossgate.nikonimglib.com/dsd_redirect/redirect.do?#{download_query[1].gsub("&amp;", "&")}"
+      merged_headers = Homebrew::Livecheck::Strategy.page_headers(redirect_url).reduce(&:merge)
+      directory = merged_headers["location"]&.match(%r{/archive7/([^/]+)/}i)
```

Comment:
```suggestion
      # Match the alphanumeric identifier from the download URL
      merged_headers = Homebrew::Livecheck::Strategy.page_headers(download_redirect_url).reduce(&:merge)
      directory = merged_headers["location"]&.[](%r{/archive\d*/([^/]+)/}i, 1)
```

There should only be one match in the `location` URL, so `&.[](..., 1)` saves us from having to use `directory[1]` later. Besides that, I've used `archive\d*` instead of `archive7`, in case the number changes (or is removed) in the future.